### ISSUE TITLE
getWords() dealing with 's

### ISSUE THREAD

**Issue by klapaukh** [UCL/PopChat-BE#24]
At the moment Lyrics.getWords() removes trailing and leading punctuation. This can cause problems for words like `goin'` and `scammin` which are used in more hip-hop style music (like the Hamilton songs). This is because the rhyming dictionary uses the trailing `'` which is removed in processing.

You can't try every word as both `word` and `word'` as in some cases both exist in the dictionary with different pronunciations. For example, the CMU dictionary we are currently using has both `goin` and `goin'`. 

Another idea would be to keep trailing punctuation if it is an `'`. However, the `'` might be part of a quoting group rather than part of a word. This is a pain because: 
 * You need to read all the text from the beginning to remove them
 * You can't tell if it is part of a quote group as words can also be truncated at the start, e.g. `'sup`, `'cos`, or `'ere`, which doesn't create a quote group with a later word like `walkin'`.

So basically I have no idea what the right answer is. The implementation currently ignores this problem, which will probably work for most songs, since the primary target is pop music where this is less common. Hence, I think not fixing this is fine for now. But it is a thing to keep in mind in future.
